### PR TITLE
overlord,store: send model assertion when setting up device sessions

### PR DESCRIPTION
**Pull request by matiasb**
When requesting (or refreshing) a device session, pass the model assertion (besides serial and session request). Store ID will be set in the session macaroon, if defined in the assertion.

### CODE REVIEW

File: overlord/auth/auth.go [hunk 1]
```diff
@@ -335,11 +335,11 @@ NextUser:
...
 	// Model returns the device model assertion.
 	Model() (*asserts.Model, error)
-	// Serial returns the device model assertion.
+	// Serial returns the device serial assertion.
 	Serial() (*asserts.Serial, error)
 
-	// DeviceSessionRequest produces a device-session-request with the given nonce, it also returns the device serial assertion.
-	DeviceSessionRequest(nonce string) (*asserts.DeviceSessionRequest, *asserts.Serial, error)
+	// DeviceSessionRequest produces a device-session-request with the given nonce, it also returns the device serial and model assertions.
+	DeviceSessionRequest(nonce string) (*asserts.DeviceSessionRequest, *asserts.Serial, *asserts.Model, error)
```
Comment (chipaca) [inline overlord/auth/auth.go]: at some point having a long list of returned values is just as confusing as having a long list of parameters, and I think it should be handled similarly by returning a struct. Personally I think three is already at the limit for this.

File: overlord/auth/auth.go [hunk 2]
```diff
@@ -448,17 +448,17 @@ func (ac *authContext) StoreID(fallback string) (string, error) {
 	return fallback, nil
 }
 
-// DeviceSessionRequest produces a device-session-request with the given nonce, it also returns the encoded device serial assertion. It returns ErrNoSerial if the device serial is not yet initialized.
-func (ac *authContext) DeviceSessionRequest(nonce string) (deviceSessionRequest []byte, serial []byte, err error) {
+// DeviceSessionRequest produces a device-session-request with the given nonce, it also returns the encoded device serial and model assertions. It returns ErrNoSerial if the device serial is not yet initialized.
+func (ac *authContext) DeviceSessionRequest(nonce string) (deviceSessionRequest []byte, serial []byte, model []byte, err error) {
```
Comment (chipaca) [inline overlord/auth/auth.go]: this is also very confusing: you have methods that are called the same, that take the same arguments and return the same number of values, that can be nil, but sometimes they're structs and sometimes they are []bytes. Reading code that uses this quickly becomes intractable.

File: overlord/auth/auth.go (hunk 1 above)
Comment (matiasb) [inline overlord/auth/auth.go]: Agreed, I can give the refactoring a try.

File: overlord/auth/auth.go (hunk 2 above)
Comment (matiasb) [inline overlord/auth/auth.go]: Also true, will also consider while refactoring.

**Pull request merged**